### ISSUE TITLE
PostgreSQL 10+ Support

### ISSUE THREAD

**Issue by LeMerP** [ClusterLabs/resource-agents#1367]
line 1923:
```
        ocf_version_cmp "$version" "10"
        rc=$?
        if [ $rc -eq 1 ]||[ $rc -eq 2 ]; then
            CHECK_XLOG_LOC_SQL="select pg_last_wal_replay_lsn(),pg_last_wal_receive_lsn()"
        else
            CHECK_XLOG_LOC_SQL="select pg_last_xlog_replay_location(),pg_last_xlog_receive_location()"
        fi
```

The current code is fixed to work only on PostgreSQL 10 and lower.
On my environment I changed the code to the following, as I do not need support for PostgreSQL lower than 9.6:
```
        ocf_version_cmp "$version" "9.6"
        rc=$?
        if [ $rc -eq 1 ]||[ $rc -eq 2 ]; then
            CHECK_XLOG_LOC_SQL="select pg_last_xlog_replay_location(),pg_last_xlog_receive_location()"
        else
            CHECK_XLOG_LOC_SQL="select pg_last_wal_replay_lsn(),pg_last_wal_receive_lsn()"
            
        fi
```

**Comment by kskmori** [comment]
I'm not sure if I understand what your actual issue is, but the current code should work on PostgreSQL 11 or later, too.

The if block is to choose proper functions for each version and intended to:
* PG version >= 10 (i.e. $rc = 1 or 2): use pg_last_wal_replay_lsn()
* PG version < 10 (i.e. $rc = 0):  use pg_last_xlog_replay_location()

The provided fix seems to work only on 9.6 and to break for other versions.

**Comment by LeMerP** [comment]
I do not looked into the function "ocf_version_cmp", so I do not know what exactly is happening here.
I was wondering why my PostgreSQL 11 resource did not came up, but PostgreSQL 10 did. Both were compiled from source on a Ubuntu 18.04 machine. With the help of pacemaker log file I could find out, that the function "pg_last_xlog_replay_location" was called instead of pg_last_wal_replay_lsn.
My suggested solution works like a charm for us for PostgreSQL 9.6, 10 and 11.

**Comment by kskmori** [comment]
It is really weird. We need to look into more details of what is happening in your environment.

Would you take the trace log of pgsql RA when it fails? See `crm resource help trace`  for how to enable trace logs (if you're using crmsh).

Also, what is the contents of $PGDATA/PG_VERSION file in your setup? My guess is that (it's only a guess though) if the PG_VERSION contains an unexpected character other than the major version number (such as "9.6" or "11" etc.) then ocf_version_cmp may return an error ($rc=3) which would lead to confusing the version determination.

**Comment by LeMerP** [comment]
Now I am very confused.
I took a deeper look into the pgsql ocf file shipped with Ubuntu 18.04.
This is the code there:
```
        ocf_version_cmp "$version" "10"
        if [ $? -eq 1 ] || [ $? -eq 2 ]; then
            CHECK_XLOG_LOC_SQL="select pg_last_wal_replay_lsn(),pg_last_wal_receive_lsn()"
        else
            CHECK_XLOG_LOC_SQL="select pg_last_xlog_replay_location(),pg_last_xlog_receive_location()"
        fi
```
You can see, that the "rc" variable is missing (now). I am very confused, because this variable is missing in my copy of pgsql ocf, too.  I really do not know what happened here.
After adding "rc=$?", the PostgreSQL 11 resource comes up like it should.

So, finally this code is working for me now ( I cannot see any difference to the code of my first comment here :/)
```
        ocf_version_cmp "$version" "10"
        rc=$?
        if [ $rc -eq 1 ] || [ $rc -eq 2 ]; then
          CHECK_XLOG_LOC_SQL="select pg_last_wal_replay_lsn(),pg_last_wal_receive_lsn()"
        else
          CHECK_XLOG_LOC_SQL="select pg_last_xlog_replay_location(),pg_last_xlog_receive_location()"
        fi
```

**Comment by kskmori** [comment]
Ah, that was a bug fixed in #1159. Sounds like a bit old version of pgsql is shipped with Ubuntu 18.04 so you would need to apply the patch manually (as you've already done).

Thank you for helping to make things clear!
